packages/slice-machine/src/modules/selectedSlice/actions.ts

```diff
@@ -71,21 +72,60 @@ export const deleteSliceWidgetMockCreator = createAction(
   newKey: string;
 }>();
 
-export const generateSliceScreenshotCreator = createAction(
-  "SLICE/GENERATE_SCREENSHOT"
-)<{ screenshots: Screenshots; component: ComponentUI }>();
+export const generateSliceScreenshotCreator = createAsyncAction(
+  "SLICE/TAKE_SCREENSHOT.REQUEST",
+  "SLICE/TAKE_SCREENSHOT.RESPONSE",
+  "SLICE/TAKE_SCREENSHOT.FAILURE"
+)<
+  {
+    _variationId: string;
+    component: ComponentUI;
+    setData: (data: any) => void;
+  },
+  { screenshots: Screenshots; component: ComponentUI }
+>();
 
-export const generateSliceCustomScreenshotCreator = createAction(
-  "SLICE/GENERATE_CUSTOM_SCREENSHOT"
-)<{ variationId: string; screenshot: ScreenshotUI; component: ComponentUI }>();
+export const generateSliceCustomScreenshotCreator = createAsyncAction(
+  "SLICE/GENERATE_CUSTOM_SCREENSHOT.REQUEST",
+  "SLICE/GENERATE_CUSTOM_SCREENSHOT.RESPONSE",
+  "SLICE/GENERATE_CUSTOM_SCREENSHOT.FAILURE"
+)<
+  {
+    variationId: string;
+    component: ComponentUI;
+    setData: (data: any) => void;
+    file: Blob;
+  },
+  { variationId: string; screenshot: ScreenshotUI; component: ComponentUI }
+>();
 
-export const saveSliceCreator = createAction("SLICE/SAVE")<{
-  extendedComponent: NonNullable<SelectedSliceStoreType>;
-}>();
+export const saveSliceCreator = createAsyncAction(
+  "SLICE/SAVE.REQUEST",
+  "SLICE/SAVE.RESPONSE",
+  "SLICE/SAVE.FAILURE"
+)<
+  {
+    extendedComponent: ExtendedComponentUI;
```

Comment:
can you change this to `NonNullable<SelectedSliceStoreType>` for consistency?
Think it's slightly better because it represents better what it is (i.e. the whole store content, except null values)